src/main/java/io/xdb/core/exception/RpcException.java

```diff
@@ -0,0 +1,8 @@
+package io.xdb.core.exception;
+
+public class RpcException extends RuntimeException {
```

Comment:
similar to the comment in server PR, we need to enrich this exception for better handling for users

Reply:
Any example or detailed explain?

Reply:
For example, the RPC implementation may want to return error that an account doesn't exist for updating an account. This error type/code need to return with a type(string) plus a message (string) so that the caller of RPC can handle the case.

Reply:
This case is already covered here: https://github.com/xcherryio/sdk-java/blob/main/src/main/java/io/xdb/core/exception/XDBHttpException.java#L54. 

The RpcException is for other rpcMethod-related error handling, not for the server error handling.

Reply:
Yeah I am mostly for SDK API level. I am not fully convinced why 

“communication.triggerNewStateExecution(StateClass, [input])” is not better than requiring a “StateDecision”. 

Particularly, the concept decision is created for state execution to make a decision. 

If you want to keep all the functionality, I would recommend that we use different APIs rather than reuse “state decision. it’s okay to reuse state decision in protocol api level but I think it’s not nice to mix in SDK level—both on concept and easy to call

Reply:
https://github.com/indeedeng/iwf/wiki/WorkflowState#statedecision-from-execute

This is how we are going to document the concept of “state decision “ and I really don’t want to mess it up and telling that — “also you can make a state decision from an RPC even there is no state”, and I really dont see a reason why we have to mess up with that 🥲

Reply:
Oops sorry, I replied in a wrong thread using my phone. I Meant to reply in the above thread for state decision.

Reply:
We can rename the state decision to process decision becuase they are really on the process level, not only on a single state level. So:

ProcessDecision -> ProcessDecision.gracefulComplete/forceComplete/... / ProcessDecision.singleNextState(...)

Reply:
Everything is on process level… state decision is to tell users that this is made from this state execution which will be different from another state execution. That’s why “next state” become meaningful because it is next based on current one.